drivers/flash/flash_shell.c

```diff
@@ -165,26 +201,37 @@ static int cmd_read(const struct shell *shell, size_t argc, char *argv[])
 static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 {
 	struct device *flash_dev;
+	size_t fa_start;
 	u32_t repeat;
+	char *endptr;
 	int result;
 	u32_t addr;
 	u32_t size;
 
-	flash_dev = device_get_binding(DT_FLASH_DEV_NAME);
-	if (!flash_dev) {
-		shell_error(shell, "Flash driver was not found!");
-		return -ENODEV;
-	}
-
-	if (argc != 4) {
-		shell_error(shell, "3 parameters reqired.");
+	if (argc < 4) {
+		shell_error(shell, "At least 3 parameters required.");
 		return -EINVAL;
 	}
+	addr = strtoul(argv[1], &endptr, 16);
+	if (*endptr != '\0') {
+		if (argc != 5) {
+			shell_error(shell, "4 parameters required.");
+			return -EINVAL;
+		}
+		flash_dev = device_get_binding(argv[1]);
```

Comment:
take a look at https://github.com/zephyrproject-rtos/zephyr/blob/b4fde3324367307b2782e2deb364670a0422d623/drivers/i2c/i2c_shell.c#L73
it is possible to have auto-completion for device name. Hopefully, this method will get into some generic device code and don't need to be copy pasted.
This should help with some potential typos when writing device name by hand.

Reply:
For flash devices, there is no common substring to exploit.  Names are set through dts labels.  

Currently, I think we could only suggest all devices, including non-flash devices.

Reply:
> Currently, I think we could only suggest all devices, including non-flash devices.

yes, exactly. It is still better than typing manually device name on each command.